lib/index.js

```diff
@@ -2,89 +2,122 @@ var request = require('request');
-	waitBetweenCalls: 200,
+  numberOfRequest: 1000,
+  statusCodeToExpect: 200, //status code to expect we can set it like a test we are expecting to get 200 but we get 400 so we have a bug
+  checkForStatusCode: true, //if set to true we can defined the check for status code and compare with the status code we are expecting to get
+  waitBetweenCalls: 200,
 };
 loadTest.counter = 1;
 loadTest.results = {
-	success: 0,
-	failure: 0,
-	unknown: 0
+  success: 0,
+  failure: 0,
+  unknown: 0
 };
 
 loadTest.init = function(options) {
-	console.log('>> Starting Load Test');
-	if (!options) {
-		throw new Error(errors.optionsIsUndefind);
-	}
+  console.log('>> Starting Load Test');
+  if (!options) {
+    throw new Error(errors.optionsIsUndefind);
+  }
 
-	if (!options.params) {
-		options.params = '';
-	}
+  if (!options.params) {
+    options.params = '';
+  }
 
-	if (!options.url) {
-		throw new Error(errors.optionsUrlIsUndefined);
-	}
+  if (!options.url) {
+    throw new Error(errors.optionsUrlIsUndefined);
+  }
 
-	if (!options.numberOfRequest) {
-		console.log('options.numberOfRequest is undefined set it to : '  + loadTest.defaultOptions.numberOfRequest);
-	}
+  if (!options.numberOfRequest) {
+    console.log('options.numberOfRequest is undefined set it to : '  + loadTest.defaultOptions.numberOfRequest);
+  }
 
-	if (!options.statusCodeToExpect) {
-			console.log('options.statusCodeToExpect is undefined set it to : '  + loadTest.defaultOptions.statusCodeToExpect);
-	}
+  if (!options.statusCodeToExpect) {
+    console.log('options.statusCodeToExpect is undefined set it to : '  + loadTest.defaultOptions.statusCodeToExpect);
+  }
 
-	loadTest.defaultOptions = _.assign(loadTest.defaultOptions, options);
+  loadTest.defaultOptions = _.assign(loadTest.defaultOptions, options);
 
-	setInterval(loadTest.createRequest, loadTest.defaultOptions.waitBetweenCalls);
+  setInterval(loadTest.createRequest, loadTest.defaultOptions.waitBetweenCalls);
 
-	console.log(loadTest.defaultOptions);
+  console.log(loadTest.defaultOptions);
 };
 
+// Makes requests at regular intervals.
+// Returns stream object which prints results.
+loadTest.requestAtInterval = function (url, optsIn) {
+  var opts = _.extend({
+    timeInterval: 200,
+    limit: 25,
+    parallel: true,
+  }, optsIn || {});
+  var wrappedRequest = hl.wrapCallback(request);
+
+  var successes = 0;
+  return hl(function (push, next) {
+    push(null, wrappedRequest(url));
+    next();
+  })
+  .errors(function (err, push) {
+    push(null, err);
+  })
+  .ratelimit(1, opts.timeInterval)
+  .take(opts.limit)
+  .through(function (stream) {
+    return opts.parallel ? stream.merge() : stream.series();
+  })
+  .on('end', function () {
+    console.log('Finished with a success rate of: ' + successes/opts.limit * 100 + '%');
+  })
+  .each(function (res) {
+    if (/2\d\d/.test(String(res.statusCode))) { successes++ }
```

Comment:
we can assume that you want 200 as a success maybe its 400 that you're expecting it should be dynamic